packages/client/src/create-deployment.ts

```diff
@@ -82,7 +86,34 @@ export default function buildCreateDeployment() {
       };
     }
 
-    const files = await hashes(fileList);
+    // Populate Files -> FileFsRef mapping
+    const workPath = typeof path === 'string' ? path : path[0];
+
+    let files;
+
+    if (clientOptions.archive === ArchiveFormat.Tgz) {
+      debug('Packing tarball');
+      fileList = fileList.map(file => file.replace(workPath, ''));
+      const tarStream = tar
+        .pack(workPath, {
+          entries: fileList,
+        })
+        .pipe(createGzip());
+      const tarBuffer: Buffer = await streamToBuffer(tarStream);
+      debug('Packed tarball');
+      files = new Map([
+        [
+          hash(tarBuffer),
```

Comment:
Do we need to worry about buffering all files in memory at once? I wonder how big this will get for large projects? We can probably optimize it later to use a streaming hash perhaps?

Reply:
Interestingly, the `hashes()` function already does buffer all files into memory, which is why the logic in `client/src/upload.ts` was able to be removed. At least in this case the hashes are thrown away for a compressed version instead 😆